### PR TITLE
fix: Notifications summary fix

### PR DESCRIPTION
**Pull request by borichellow**
----
## [Bug-2019](https://wearezeta.atlassian.net/browse/AR-2019)
----

### Issues
Sometimes notifications from the app are not grouped into one big notification

### Causes (Optional)

The problem was that summaryNotification for the messages and callNotification had the same ID. 
🤦 
So when there were some notifications and call comes -> callNotification replaces summaryNotification -> notifications are not grouped anymore.

### Solutions

Created `NotificationConstants` object which stores all the notification consts and can be sure that there are no repeats.
Also added some logs to have more info in the future.

#### How to Test

- get some message notifications from different conversations
- see that notifications are grouped
- receive a call (when the app is in a background or covered by another app, so the callNotification comes)
- decline a call, or wait until the caller cancels it
- See that message notifications are still there and grouped

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-782/1/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3928026)(3928026)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3928026)(3928026)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/notification/MessageNotificationManager.kt [hunk 1]
```diff
@@ -68,14 +78,18 @@ class MessageNotificationManager @Inject constructor(private val context: Contex
...
         newData: List<LocalNotificationConversation>,
         userId: String?
     ) {
-        if (newData.isEmpty()) notificationManager.cancel(SUMMARY_ID)
-
-        if (oldData.size <= 1 && newData.size > 1)
+        if (newData.isEmpty()) {
+            appLogger.i("$TAG removing groupSummary")
+            notificationManager.cancel(SUMMARY_ID)
+            isGroupSummaryDisplayed = false
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/notification/MessageNotificationManager.kt:87]: What if this notification is swiped out by user? Then it should also be changed to "false", right?

**Comment by saleniuk** [comment]
I found out that both summary notification and call notification have ID == 0, maybe that's an issue?
I usually create a resource of `type="id"` in xml for each notification, this way every single one is unique and constant 😄

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-782/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-782/3/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3943986)(3943986)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3943986)(3943986)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-782/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3944177)(3944177)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3944177)(3944177)-dev-release.apk)

**Review by yamilmedina — APPROVED** [review]
lgtm! Nice to see more logs, they always help 🪄

**Pull request merged**